tests/hierarchy_driver.cc

```diff
@@ -80,38 +163,198 @@ int main(int argc, char *argv[])
 {
   namespace boost_po = boost::program_options;
 
-  using f_type = std::vector<std::string>;
-
   MPI_Init(&argc, &argv);
 
   boost_po::options_description cmd("Available options");
-  cmd.add_options()("filename,f", boost_po::value<f_type>()->multitoken(),
-                    "file(s) containing input parameters");
+  cmd.add_options()("help,h", "produce help message");
+  cmd.add_options()("filename,f", boost_po::value<std::string>()->multitoken(),
+                    "file containing input parameters");
   cmd.add_options()("dim,d", boost_po::value<int>(), "dimension");
+  cmd.add_options()("matrix_free,m", boost_po::value<bool>(),
+                    "use matrix-free algorithm");
```

Comment:
Why did you choose `value<bool>()` over `bool_switch()`?

Reply:
no reason